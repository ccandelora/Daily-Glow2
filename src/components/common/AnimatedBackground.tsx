import React, { useEffect, useRef } from 'react';
import { Animated, StyleSheet, View, Dimensions } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import theme from '@/constants/theme';

interface AnimatedBackgroundProps {
  intensity?: 'light' | 'medium' | 'strong';
  animated?: boolean;
}

export const AnimatedBackground: React.FC<AnimatedBackgroundProps> = ({ 
  intensity = 'medium',
  animated = true 
}) => {
  const fadeAnim = useRef(new Animated.Value(0)).current;
  const rotateAnim = useRef(new Animated.Value(0)).current;

  useEffect(() => {
    if (animated) {
      // Fade in
      Animated.timing(fadeAnim, {
        toValue: 1,
        duration: 1000,
        useNativeDriver: true,
      }).start();

      // Faster rotation
      const rotateAnimation = Animated.loop(
        Animated.sequence([
          Animated.timing(rotateAnim, {
            toValue: 1,
            duration: 30000, // 30 seconds per rotation
            useNativeDriver: true,
          }),
          Animated.timing(rotateAnim, {
            toValue: 0,
            duration: 0,
            useNativeDriver: true,
          })
        ])
      );

      rotateAnimation.start();
    } else {
      fadeAnim.setValue(1);
    }
  }, [animated]);

  const spin = rotateAnim.interpolate({
    inputRange: [0, 1],
    outputRange: ['0deg', '360deg']
  });

  return (
    <View style={styles.container}>
      {/* Static base gradient */}
      <LinearGradient
        colors={[
          'rgba(255,255,255,0.92)',
          'rgba(235,230,255,0.95)',
          'rgba(255,255,255,0.92)',
        ]}
        style={styles.baseGradient}
        start={{ x: 0, y: 0 }}
        end={{ x: 0, y: 1 }}
      />

      <Animated.View
        style={[
          styles.gradientContainer,
          {
            opacity: fadeAnim,
            transform: [{ rotate: spin }]
          }
        ]}
      >
        {/* Brighter purple glow */}
        <LinearGradient
          colors={[
            'transparent',
            `${theme.COLORS.primary.blue}15`,
            `${theme.COLORS.primary.blue}20`,
            `${theme.COLORS.primary.blue}15`,
            'transparent',
          ]}
          style={styles.gradient}
          start={{ x: 0.3, y: 0.3 }}
          end={{ x: 0.7, y: 0.7 }}
          locations={[0, 0.3, 0.5, 0.7, 1]}
        />

        {/* Stronger accent highlights */}
        <LinearGradient
          colors={[
            'transparent',
            'rgba(230,230,255,0.35)',
            'transparent'
          ]}
          style={styles.gradient}
          start={{ x: 0, y: 0.5 }}
          end={{ x: 1, y: 0.5 }}
        />

        {/* Enhanced color variation */}
        <LinearGradient
          colors={[
            'transparent',
            `${theme.COLORS.primary.blue}12`,
            'transparent'
          ]}
          style={styles.gradient}
          start={{ x: 0.5, y: 0 }}
          end={{ x: 0.5, y: 1 }}
        />
      </Animated.View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    backgroundColor: 'white',
  },
  baseGradient: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
  },
  gradientContainer: {
    position: 'absolute',
    top: -400,
    left: -400,
    right: -400,
    bottom: -400,
  },
  gradient: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    transform: [{ scale: 2 }],
  },
}); 