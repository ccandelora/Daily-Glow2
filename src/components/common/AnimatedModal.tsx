import React, { useEffect } from 'react';
import {
  View,
  Modal,
  StyleSheet,
  TouchableOpacity,
  Animated,
  Dimensions,
  ViewStyle,
} from 'react-native';
import theme from '../../constants/theme';
import { Typography } from './Typography';

interface AnimatedModalProps {
  visible: boolean;
  onClose: () => void;
  title?: string;
  children: React.ReactNode;
  style?: ViewStyle;
}

export const AnimatedModal: React.FC<AnimatedModalProps> = ({
  visible,
  onClose,
  title,
  children,
  style,
}) => {
  const slideAnim = React.useRef(new Animated.Value(0)).current;
  const fadeAnim = React.useRef(new Animated.Value(0)).current;

  useEffect(() => {
    if (visible) {
      Animated.parallel([
        Animated.timing(slideAnim, {
          toValue: 1,
          duration: 300,
          useNativeDriver: true,
        }),
        Animated.timing(fadeAnim, {
          toValue: 1,
          duration: 300,
          useNativeDriver: true,
        }),
      ]).start();
    } else {
      Animated.parallel([
        Animated.timing(slideAnim, {
          toValue: 0,
          duration: 200,
          useNativeDriver: true,
        }),
        Animated.timing(fadeAnim, {
          toValue: 0,
          duration: 200,
          useNativeDriver: true,
        }),
      ]).start();
    }
  }, [visible, slideAnim, fadeAnim]);

  const { height } = Dimensions.get('window');

  return (
    <Modal
      transparent
      visible={visible}
      onRequestClose={onClose}
      animationType="none"
    >
      <Animated.View
        style={[
          styles.overlay,
          {
            opacity: fadeAnim,
          },
        ]}
      >
        <TouchableOpacity
          style={styles.backdrop}
          onPress={onClose}
          activeOpacity={1}
        />
        <Animated.View
          style={[
            styles.content,
            {
              transform: [
                {
                  translateY: slideAnim.interpolate({
                    inputRange: [0, 1],
                    outputRange: [height, 0],
                  }),
                },
              ],
            },
            style,
          ]}
        >
          {title && (
            <View style={styles.header}>
              <Typography variant="h3">{title}</Typography>
              <TouchableOpacity onPress={onClose} style={styles.closeButton}>
                <Typography variant="h3">Ã—</Typography>
              </TouchableOpacity>
            </View>
          )}
          {children}
        </Animated.View>
      </Animated.View>
    </Modal>
  );
};

const styles = StyleSheet.create({
  overlay: {
    flex: 1,
    backgroundColor: 'rgba(0, 0, 0, 0.5)',
    justifyContent: 'flex-end',
  },
  backdrop: {
    ...StyleSheet.absoluteFillObject,
  },
  content: {
    backgroundColor: theme.COLORS.ui.background,
    borderTopLeftRadius: theme.BORDER_RADIUS.xl,
    borderTopRightRadius: theme.BORDER_RADIUS.xl,
    padding: theme.SPACING.lg,
    maxHeight: '90%',
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: theme.SPACING.md,
  },
  closeButton: {
    padding: theme.SPACING.sm,
    margin: -theme.SPACING.sm,
  },
}); 