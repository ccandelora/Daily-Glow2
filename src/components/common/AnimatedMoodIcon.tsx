import React, { useEffect } from 'react';
import { Animated, StyleSheet, ViewStyle } from 'react-native';
import theme from '../../constants/theme';

interface AnimatedMoodIconProps {
  size?: number;
  color: string;
  style?: ViewStyle;
  children: React.ReactNode;
  active?: boolean;
}

export const AnimatedMoodIcon: React.FC<AnimatedMoodIconProps> = ({
  size = 60,
  color,
  style,
  children,
  active = false,
}) => {
  const scaleAnim = React.useRef(new Animated.Value(1)).current;
  const opacityAnim = React.useRef(new Animated.Value(0.6)).current;

  useEffect(() => {
    Animated.parallel([
      Animated.spring(scaleAnim, {
        toValue: active ? 1.1 : 1,
        useNativeDriver: true,
        damping: 10,
      }),
      Animated.timing(opacityAnim, {
        toValue: active ? 1 : 0.6,
        duration: 200,
        useNativeDriver: true,
      }),
    ]).start();
  }, [active, scaleAnim, opacityAnim]);

  return (
    <Animated.View
      style={[
        styles.container,
        {
          width: size,
          height: size,
          backgroundColor: color,
          transform: [{ scale: scaleAnim }],
          opacity: opacityAnim,
        },
        style,
      ]}
    >
      {children}
    </Animated.View>
  );
};

const styles = StyleSheet.create({
  container: {
    borderRadius: theme.BORDER_RADIUS.circle,
    alignItems: 'center',
    justifyContent: 'center',
    overflow: 'hidden',
  },
}); 