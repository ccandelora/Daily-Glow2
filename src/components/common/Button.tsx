import React from 'react';
import { TouchableOpacity, StyleSheet, ViewStyle, TextStyle, ActivityIndicator, StyleProp } from 'react-native';
import { Typography } from './Typography';
import theme from '@/constants/theme';

interface ButtonProps {
  title: string;
  onPress: () => void;
  variant?: 'primary' | 'secondary' | 'outline';
  size?: 'small' | 'medium' | 'large';
  disabled?: boolean;
  loading?: boolean;
  style?: ViewStyle;
  textStyle?: TextStyle;
}

export const Button: React.FC<ButtonProps> = ({
  title,
  onPress,
  variant = 'primary',
  size = 'medium',
  disabled = false,
  loading = false,
  style,
  textStyle,
}) => {
  return (
    <TouchableOpacity
      style={[
        styles.base,
        styles[variant],
        styles[size],
        disabled && styles.disabled,
        style,
      ]}
      onPress={onPress}
      disabled={disabled || loading}
    >
      {loading ? (
        <ActivityIndicator
          color={variant === 'outline' ? theme.COLORS.primary.green : theme.COLORS.ui.background}
          size="small"
        />
      ) : (
        <Typography
          variant="button"
          style={[
            styles.text,
            styles[`${variant}Text` as keyof typeof styles] as TextStyle,
            disabled && styles.disabledText,
            textStyle,
          ].filter(Boolean) as TextStyle}
        >
          {title}
        </Typography>
      )}
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  base: {
    borderRadius: theme.BORDER_RADIUS.lg,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
    shadowColor: theme.COLORS.ui.accent,
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.3,
    shadowRadius: 8,
    elevation: 4,
  },
  primary: {
    backgroundColor: theme.COLORS.ui.accent,
    borderWidth: 0,
    shadowColor: theme.COLORS.ui.accent,
    shadowOpacity: 0.5,
    shadowRadius: 12,
  },
  secondary: {
    backgroundColor: 'rgba(65, 105, 225, 0.1)',
    borderWidth: 1,
    borderColor: theme.COLORS.ui.accent,
  },
  outline: {
    backgroundColor: 'transparent',
    borderWidth: 1,
    borderColor: theme.COLORS.ui.border,
  },
  small: {
    paddingVertical: theme.SPACING.sm,
    paddingHorizontal: theme.SPACING.lg,
    minWidth: 100,
  },
  medium: {
    paddingVertical: theme.SPACING.md,
    paddingHorizontal: theme.SPACING.xl,
    minWidth: 150,
  },
  large: {
    paddingVertical: theme.SPACING.lg,
    paddingHorizontal: theme.SPACING.xl * 1.5,
    minWidth: 200,
  },
  disabled: {
    backgroundColor: theme.COLORS.ui.disabled,
    borderColor: 'transparent',
    shadowOpacity: 0,
  },
  text: {
    textAlign: 'center',
  },
  primaryText: {
    color: theme.COLORS.ui.text,
    textShadowColor: 'rgba(0, 0, 0, 0.2)',
    textShadowOffset: { width: 0, height: 1 },
    textShadowRadius: 2,
  },
  secondaryText: {
    color: theme.COLORS.ui.accent,
    textShadowColor: theme.COLORS.ui.accent,
    textShadowOffset: { width: 0, height: 0 },
    textShadowRadius: 4,
  },
  outlineText: {
    color: theme.COLORS.ui.text,
  },
  disabledText: {
    color: theme.COLORS.ui.textSecondary,
  },
}); 