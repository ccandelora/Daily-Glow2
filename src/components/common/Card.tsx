import React from 'react';
import { View, StyleSheet, ViewStyle } from 'react-native';
import theme from '../../constants/theme';

interface CardProps {
  children: React.ReactNode;
  style?: ViewStyle;
  variant?: 'default' | 'elevated';
}

export const Card: React.FC<CardProps> = ({
  children,
  style,
  variant = 'default',
}) => {
  return (
    <View
      style={[
        styles.card,
        variant === 'elevated' && styles.elevated,
        style,
      ]}
    >
      {children}
    </View>
  );
};

const styles = StyleSheet.create({
  card: {
    backgroundColor: theme.COLORS.ui.card,
    borderRadius: theme.BORDER_RADIUS.lg,
    padding: theme.SPACING.md,
    marginVertical: theme.SPACING.sm,
  },
  elevated: {
    shadowColor: theme.COLORS.ui.text,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
}); 