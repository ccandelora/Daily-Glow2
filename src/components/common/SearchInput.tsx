import React from 'react';
import { TextInput, View, StyleSheet, TextInputProps, TouchableOpacity } from 'react-native';
import { FontAwesome6 } from '@expo/vector-icons';
import theme from '@/constants/theme';

interface SearchInputProps extends TextInputProps {
  value: string;
  onChangeText: (text: string) => void;
  onClear?: () => void;
}

export const SearchInput: React.FC<SearchInputProps> = ({
  value,
  onChangeText,
  onClear,
  placeholder = "Search...",
  style,
  ...props
}) => {
  return (
    <View style={[styles.container, style]}>
      <FontAwesome6 
        name="search" 
        size={18} 
        color={theme.COLORS.ui.textSecondary}
        style={styles.searchIcon}
      />
      <TextInput
        style={styles.input}
        value={value}
        onChangeText={onChangeText}
        placeholder={placeholder}
        placeholderTextColor={theme.COLORS.ui.textSecondary}
        autoCapitalize="none"
        autoCorrect={false}
        clearButtonMode="never"
        returnKeyType="search"
        {...props}
      />
      {value ? (
        <TouchableOpacity
          onPress={onClear}
          style={styles.clearButton}
          hitSlop={{ top: 10, right: 10, bottom: 10, left: 10 }}
        >
          <FontAwesome6 
            name="circle-xmark" 
            size={18} 
            color={theme.COLORS.ui.textSecondary} 
          />
        </TouchableOpacity>
      ) : null}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: `${theme.COLORS.ui.card}80`,
    borderRadius: theme.BORDER_RADIUS.md,
    paddingHorizontal: theme.SPACING.md,
    borderWidth: 1,
    borderColor: `${theme.COLORS.ui.border}50`,
    height: 50,
  },
  searchIcon: {
    marginRight: theme.SPACING.sm,
  },
  input: {
    flex: 1,
    height: '100%',
    color: theme.COLORS.ui.text,
    fontSize: 16,
    paddingVertical: 0,
  },
  clearButton: {
    padding: theme.SPACING.xs,
    marginLeft: theme.SPACING.xs,
  },
}); 