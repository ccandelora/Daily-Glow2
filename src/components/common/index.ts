export * from './Button';
export * from './Card';
export * from './Typography';
export * from './Input';
export * from './ProgressBar';
export * from './Header';
export * from './AnimatedModal';
export * from './AnimatedMoodIcon';
export * from './EmotionWheel';
export { LoadingSpinner } from './LoadingSpinner';
export { Toast } from './Toast';
export { LoadingOverlay } from './LoadingOverlay';
export { DailyChallenge } from './DailyChallenge';
export { NotificationBadge } from './NotificationBadge';
export { TabBar } from './TabBar'; 