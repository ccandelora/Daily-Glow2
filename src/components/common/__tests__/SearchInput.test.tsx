import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { SearchInput } from '@/components/common/SearchInput';
import { TextInput, TouchableOpacity } from 'react-native';

// Mock FontAwesome6 from @expo/vector-icons
jest.mock('@expo/vector-icons', () => ({
  FontAwesome6: 'FontAwesome6'
}));

describe('SearchInput Component', () => {
  const mockOnChangeText = jest.fn();
  const mockOnClear = jest.fn();
  
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with default props', () => {
    const { toJSON, getByPlaceholderText } = render(
      <SearchInput value="" onChangeText={mockOnChangeText} />
    );
    
    // Should have rendered a TextInput with the correct placeholder
    expect(getByPlaceholderText('Search...')).toBeTruthy();
    
    // Should match snapshot
    expect(toJSON()).toMatchSnapshot();
  });

  it('passes the value to the TextInput', () => {
    const testValue = 'test search';
    const { getByDisplayValue } = render(
      <SearchInput value={testValue} onChangeText={mockOnChangeText} />
    );
    
    // Input should have the provided value
    expect(getByDisplayValue(testValue)).toBeTruthy();
  });

  it('calls onChangeText when text changes', () => {
    const { getByPlaceholderText } = render(
      <SearchInput value="" onChangeText={mockOnChangeText} />
    );
    
    const inputElement = getByPlaceholderText('Search...');
    const newValue = 'new search text';
    
    fireEvent.changeText(inputElement, newValue);
    
    // onChangeText should be called with the new value
    expect(mockOnChangeText).toHaveBeenCalledWith(newValue);
  });

  it('displays the clear button when value is not empty', () => {
    const { UNSAFE_getAllByType } = render(
      <SearchInput value="test" onChangeText={mockOnChangeText} onClear={mockOnClear} />
    );
    
    // Should have a TouchableOpacity for clearing
    const touchables = UNSAFE_getAllByType(TouchableOpacity);
    expect(touchables.length).toBe(1); // One clear button
  });

  it('does not display the clear button when value is empty', () => {
    const { UNSAFE_queryAllByType } = render(
      <SearchInput value="" onChangeText={mockOnChangeText} onClear={mockOnClear} />
    );
    
    // Should not have a TouchableOpacity for clearing
    const touchables = UNSAFE_queryAllByType(TouchableOpacity);
    expect(touchables.length).toBe(0); // No clear button
  });

  it('calls onClear when clear button is pressed', () => {
    const { UNSAFE_getAllByType } = render(
      <SearchInput value="test" onChangeText={mockOnChangeText} onClear={mockOnClear} />
    );
    
    const clearButton = UNSAFE_getAllByType(TouchableOpacity)[0];
    fireEvent.press(clearButton);
    
    // onClear should be called
    expect(mockOnClear).toHaveBeenCalledTimes(1);
  });

  it('allows custom placeholder text', () => {
    const customPlaceholder = 'Find something...';
    const { getByPlaceholderText } = render(
      <SearchInput value="" onChangeText={mockOnChangeText} placeholder={customPlaceholder} />
    );
    
    // Should use the custom placeholder
    expect(getByPlaceholderText(customPlaceholder)).toBeTruthy();
  });

  it('applies custom style to container', () => {
    const customStyle = { backgroundColor: 'red' };
    const { toJSON } = render(
      <SearchInput value="" onChangeText={mockOnChangeText} style={customStyle} />
    );
    
    // Get the snapshot and check it contains our custom style
    const snapshot = toJSON();
    expect(snapshot).toMatchSnapshot();
  });
}); 