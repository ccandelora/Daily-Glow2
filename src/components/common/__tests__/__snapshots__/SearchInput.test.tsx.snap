// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput Component applies custom style to container 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "rgba(38, 20, 60, 0.85)80",
        "borderColor": "rgba(82, 67, 194, 0.3)50",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 50,
        "paddingHorizontal": 16,
      },
      {
        "backgroundColor": "red",
      },
    ]
  }
>
  <FontAwesome6
    color="rgba(255, 255, 255, 0.7)"
    name="search"
    size={18}
    style={
      {
        "marginRight": 8,
      }
    }
  />
  <TextInput
    autoCapitalize="none"
    autoCorrect={false}
    clearButtonMode="never"
    onChangeText={[MockFunction]}
    placeholder="Search..."
    placeholderTextColor="rgba(255, 255, 255, 0.7)"
    returnKeyType="search"
    style={
      {
        "color": "#FFFFFF",
        "flex": 1,
        "fontSize": 16,
        "height": "100%",
        "paddingVertical": 0,
      }
    }
    value=""
  />
</View>
`;

exports[`SearchInput Component renders correctly with default props 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "rgba(38, 20, 60, 0.85)80",
        "borderColor": "rgba(82, 67, 194, 0.3)50",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 50,
        "paddingHorizontal": 16,
      },
      undefined,
    ]
  }
>
  <FontAwesome6
    color="rgba(255, 255, 255, 0.7)"
    name="search"
    size={18}
    style={
      {
        "marginRight": 8,
      }
    }
  />
  <TextInput
    autoCapitalize="none"
    autoCorrect={false}
    clearButtonMode="never"
    onChangeText={[MockFunction]}
    placeholder="Search..."
    placeholderTextColor="rgba(255, 255, 255, 0.7)"
    returnKeyType="search"
    style={
      {
        "color": "#FFFFFF",
        "flex": 1,
        "fontSize": 16,
        "height": "100%",
        "paddingVertical": 0,
      }
    }
    value=""
  />
</View>
`;
